src/FileFormats/SDPA/SDPA.jl

```diff
@@ -251,14 +268,28 @@ function Base.read!(io::IO, model::Model{T}) where T
         if startswith(line, '"')
             continue
         end
+        # The lines starting with * should also be skipped 
+        # according to http://plato.asu.edu/ftp/sdpa_format.txt.
+        if startswith(line, '*')
+            # Exceptions for integer variables
+            if startswith(line, "*INTEGER")
+                integer_read = true
+            elseif integer_read
+                if !num_variables_read
+                    error("The number of variables should be given before *INTEGER section.")
+                end
+                push!(intvar_idx, parse(Int, strip(line[2:end])))
+            end
+            continue
+        end
         if !num_variables_read
             if isempty(line)
                 continue
             end
             num_variables_read = true
             # According to http://plato.asu.edu/ftp/sdpa_format.txt,
             # additional text after the number of variables should be ignored.
-            MOI.add_variables(model, parse(Int, split(line)[1]))
+            scalar_vars = MOI.add_variables(model, parse(Int, split(line)[1]))
```

Comment:
Can you modify `MOI.VariableIndex(matrix)`  below into `scalars_vars[matrix]`?

Reply:
Both suggestions were handled and passed. Thanks!